static/js/redux/ui/search_bar.jsx

```diff
@@ -62,11 +62,17 @@ class SearchBar extends React.Component {
         if (e.key === 'Enter' && numSearchResults > 0 && this.state.showDropdown) {
           this.props.addCourse(this.props.searchResults[this.props.hoveredPosition].id);
         } else if (e.key === 'ArrowDown') {
-          this.props.hoverSearchResult((this.props.hoveredPosition + 1) % numSearchResults);
+          let newHoveredPosition = this.props.hoveredPosition;
+          newHoveredPosition = newHoveredPosition < numSearchResults-1 ? newHoveredPosition+1 : newHoveredPosition;
+          this.props.hoverSearchResult(newHoveredPosition);
+          if($('#results_scroll').scrollTop()<$('#results_scroll').prop("scrollHeight")/numSearchResults*(this.props.hoveredPosition-3))
```

Comment:
can you split this up and add comments? its hard to follow
i.e. make the right side of the < a var declared above

Reply:
also i looked into getting rid of the jQuery and i don't think there's another good way of doing it so lets just keep it